llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -863,15 +863,15 @@ void WaitcntBrackets::updateByEvent(const SIInstrInfo *TII,
         // can be used as the actual source after export patching, so
         // we need to treat them like sources and set the EXP_CNT
         // score.
-        for (MachineOperand &DefMO : Inst.all_defs()) {
+        for (MachineOperand &DefMO : Inst.defs()) {
           if (TRI->isVGPR(*MRI, DefMO.getReg())) {
             setRegScore(
                 TRI->getEncodingValue(AMDGPU::getMCReg(DefMO.getReg(), *ST)),
                 EXP_CNT, CurrScore);
           }
         }
       }
-      for (const MachineOperand &Op : Inst.all_uses()) {
+      for (const MachineOperand &Op : Inst.uses()) {
```

Comment:
Here it's not obvious (to me) what kind of instruction `Inst` is.

Reply:
We do not define implicit VGPR uses anywhere, except for super-reg liveness purposes. This is my assumption here. The next line checks it is a VGPR, so this is not some of the special registers. Am I wrong in that assumption?